server_proxy/src/pxmhub.c

```diff
@@ -3717,12 +3727,18 @@ static int mhubReceiveMetadata(ism_transport_t * transport, char * inbuf, int bu
         mhub->expectingMetadata = 0;
     }
     if (rc) {
+    	LOG(ERROR, Server, 983, "%u%s%s%u%s", "Mhub Metadata is incomplete: connect={0} name={1} server_addr={2} server_port={3} broker={4}",
+								transport->index, transport->name, transport->server_addr, transport->serverport,
+											(mhub->trybroker>0)?mhub->brokers[mhub->trybroker-1]:mhub->brokers[0]);
         transport->close(transport, ISMRC_BadClientData, 0, "MessageHub metadata incomplete");
 
         ism_mhub_lock(mhub);
-        if(!g_shuttingDown && mhub->enabled==1)
-        		ism_common_setTimerOnce(ISM_TIMER_LOW, (ism_attime_t)mhubRetryConnect, mhub, retryDelay(mhub->retry++));
-        ism_mhub_unlock(mhub);
+		if(!g_shuttingDown && mhub->enabled==1){
+			ism_mhub_unlock(mhub);
```

Comment:
I'm not sure about this, don't we want to hold the lock until after we have set the timer?

Reply:
think the version added in line 3800 also needs to change

Reply:
Updated the code based on the comments